### PR TITLE
chore(installer): build nsis installation script

### LINKED ISSUE #1: Investigate ui <> core communication strategies
- Investigate how to pack and publish the whole project (core + desktop), build a small PoC

https://www.youtube.com/watch?v=Z8qCn32tXUo

https://github.com/morochena/rails-in-electron

### PR DESCRIPTION
**Pull request by mapra99**
Closes #1
This PR builds an initial version for the installation build of the Structuraid project. It does the following:
- It builds the desktop app and daemon app
- It copies the build files into the installation directory (Program Files, by default)
- It creates shortcuts for the desktop app, registers the app, and creates an uninstaller

The readme is updated with instructions to do the process. I created a bash script that needs to be run before compiling the actual NSIS script.

I think this feels like a good starting point, but I also created some follow-up items to investigate them separately:
- #2
- #3
- #4
- #6

### CODE REVIEW

File: src/install-script.nsi [hunk 1]
```diff
@@ -0,0 +1,417 @@
...
+Name "${PRODUCT_NAME} ${PRODUCT_VERSION}"
+OutFile "Structuraid.exe"
+InstallDir "$PROGRAMFILES64\Structuraid"
+InstallDirRegKey HKLM "${PRODUCT_DIR_REGKEY}" ""
+ShowInstDetails show
+ShowUnInstDetails show
+
+Section "MainSection" SEC01
+  SetOutPath "$INSTDIR"
+  File "..\dist\.keep"
```
Comment (mapra99) [inline src/install-script.nsi:50]: This is copying every single file in the `dist/` folder, 1 by 1. Not sure if there is a way to copy the whole folder in a single command 😬

Comment (mapra99) [reply src/install-script.nsi:50]: Looks like it's possible. Will test this out later
https://stackoverflow.com/questions/11014013/copy-a-directory-using-nsis

**Review by ImMPrada — APPROVED** [review]
nice!

File: src/install-script.nsi (hunk 1 above)
Comment (ImMPrada) [reply src/install-script.nsi:50]: 🥹

**Pull request merged**